go/base/context.go

```diff
@@ -135,6 +135,7 @@ type MigrationContext struct {
 	CriticalLoadHibernateSeconds        int64
 	PostponeCutOverFlagFile             string
 	CutOverLockTimeoutSeconds           int64
+	CutOverIdleTimeoutSeconds           int64
```

Comment:
Since this value must be greater than `CutOverLockTimeoutSeconds` (or else cut-over may timeout unjustifiably), I suggest removing it altogether and setting something like `CutOverLockTimeoutSeconds * 2`.

Reply:
@shlomi-noach a `CutOverLockTimeoutSeconds * 2` approach is what I took initially, and I think it will probably work in practice, but I may have got a bit paranoid about one detail that I hope you can validate

The `CutOverLockTimeoutSeconds` (== `lock_wait_timeout`) tells MySQL how long the lock can be held once it is taken (3 seconds by default), but nothing is controlling how long the "whole operation" takes to issue. So if we use `CutOverLockTimeoutSeconds * 2` as the `wait_timeout`, let's call this `6`, we are assuming the `RENAME` will always take < `6` seconds, `3` seconds max locking, `3` seconds max in "processing"/network time

So what if, for whatever crazy reason (we are in this kind of territory now with this PR 😄) the server (or network) is slow to process the `RENAME` operation in the 2nd connection. The connection holding the table locks (with `wait_timeout=6`) may close at `6` seconds and in some extremely-rare edge cases this may be too early. Can you help validate this idea?

This paranoia led me to make the `wait_timeout` configurable (still doesn't solve every case) and to make sure the "waiting" connection keeps it's connection "warm" - in case the `RENAME` is slow

Reply:
كاني نايك امك

Reply:
@shlomi-noach I agree the simpler approach would be better, at least for now. Thanks for validating this highly-theoretical risk! I'll make this update

Reply:
Removed `CutOverIdleTimeoutSeconds` 👍